### PR TITLE
Add *lots of icons*

### LINKED ISSUE #496: Icon pack proposal: Entypo+
- Name: `Entypo+`
- License: `CC BY-SA 4.0`
- NPM package: [entypo](https://github.com/hypermodules/entypo)
- Website (if any): http://www.entypo.com/

Preview

### LINKED ISSUE #497: Icon pack proposal: Zondicons
- Name: Zondicons
- License: `CC-BY-4.0`
- NPM package: [zondicons](https://www.npmjs.com/package/zondicons)
- Website (if any): https://www.zondicons.com

Preview

### LINKED ISSUE #499: Icon pack proposal: Open Iconic
- Name: `Open Iconic`
- License: `MIT License`
- NPM package: [open-iconic](https://www.npmjs.com/package/open-iconic)
- Website (if any): https://useiconic.com/open/

Preview

### LINKED ISSUE #501: Icon pack proposal: Foundation Icons
- Name: `Foundation Icons`
- License: `MIT`
- NPM package: [@icon/foundation-icons](https://www.npmjs.com/package/@icon/foundation-icons)
- Website (if any): https://zurb.com/playground/foundation-icon-fonts-3

image

### LINKED ISSUE #1166: Icon pack: Heroicons
- Name: `Heroicons`
- License: `MIT`
- NPM package: `heroicons`
- Website (if any): `https://github.com/refactoringui/heroicons`

Used for Tailwind UI

### PR DESCRIPTION
**Pull request by jacobwgillespie**
These are all outstanding icon pack requests - let's see how the build handles this 😛 

Closes #1166
Closes #501 
Closes #499 
Closes #497 
Closes #496

### CODE REVIEW

**Comment by stramel** [comment]
Guess I should rebase my material variants PR... see if I can get that to work with the latest changes.

**Comment by stramel** [comment]
@jacobwgillespie Searching seems broken

**Comment by stramel** [comment]
Scrolling also breaks after getting 1/4 way down the page

**Comment by jacobwgillespie** [comment]
> Seems to break the website

Yeah I was going to be really surprised if it worked the first time 😛 

I believe the issue is I haven't added the new icon packs to the website yet, although they are in the manifest JSON file the website reads from, so it's trying to import packages it doesn't have in the bundle.

That said, I may need to implement a tiered import scheme, where each pack is imported individually as a dynamic `import()`, so that it breaks up the size of the JS bundle and perhaps the compile burden too.  I think if I just add the packs directly to what we have, it'll probably trigger the CI OOM issue again.

**Comment by stramel** [comment]
@jacobwgillespie FWIW,  I have tried breaking it up like that previously and the build still hung on me. Feel free to give it a shot though!

File: packages/website/src/components/IconExplorer.tsx [hunk 1]
```diff
@@ -65,6 +78,15 @@ const icons = iconManifest.map(
       case 'feather':
         return {...icon, icon: feather[icon.name as keyof typeof feather]}
 
+      case 'founcation':
```
Comment (stramel) [inline packages/website/src/components/IconExplorer.tsx]:
```suggestion
      case 'foundation':
```

Comment (jacobwgillespie) [reply packages/website/src/components/IconExplorer.tsx]: Nice :)

**Comment by stramel** [comment]
Seems to be failing when searching `fou`...

**Comment by jacobwgillespie** [comment]
Hey, it built! 🎉 It appears to have packaged the app into one chunk per icon pack too, which is cool.  Not sure if it's actually loading the app before all the chunks are loaded or not, but this seems safe to deploy for now.

**Comment by stramel** [comment]
I will do some verification when I get home tonight. I believe the imports are incorrect at first glance.

I have a few other ideas to improve the load but I can build on this if this is working.

**Comment by jacobwgillespie** [comment]
> I think they're supposed to be `() => import('')`

It looks like it's working as intended, though it might be possible to be optimized further.  Right now, it loads the app and when it loads the chunk with the dynamic import object, it immediately queues all the requests for all the icon packs.  `import()` returns a promise and isn't blocking - the app does initialize and load before the packs are loaded, and once the packs are loaded, the icons visually load.

I'm not sure what the behavior would look like if it only loaded visible packs - that might be an interesting optimization.  For that, we would want `() => import()` to defer loading the packs.

**Comment by Betree** [comment]
Thank you for adding all that @jacobwgillespie! :tada:

**Pull request merged**